client/my-sites/checklist/checklist-show/index.jsx

```diff
@@ -38,49 +41,57 @@ class ChecklistShow extends PureComponent {
 		} );
 	};
 
-	handleToggle = id => {
-		const { notify, siteId, tasks, update } = this.props;
-		const task = find( tasks, { id } );
+	handleTaskDismiss = task => () => {
+		const { notify, siteId, update } = this.props;
 
 		if ( task && ! task.completed ) {
 			notify( 'is-success', 'You completed a task!' );
-			update( siteId, id );
+			update( siteId, task.id );
 		}
 	};
 
 	render() {
-		const { siteId, tasks } = this.props;
+		const { isJetpack, siteId, taskStatuses } = this.props;
+		const tasks = isJetpack ? jetpackTasks : wpcomTasks;
 
 		return (
 			<Fragment>
 				{ siteId && <QuerySiteChecklist siteId={ siteId } /> }
-				<Checklist
-					isLoading={ ! tasks }
-					tasks={ tasks }
-					onAction={ this.handleAction }
-					onToggle={ this.handleToggle }
-				/>
+				<Checklist isPlaceholder={ ! taskStatuses }>
+					{ tasks.map( task => (
+						<Task
+							buttonPrimary={ task.buttonPrimary }
+							buttonText={ task.buttonText }
+							completed={ task.completed || get( taskStatuses, [ task.id, 'completed' ], false ) }
+							completedButtonText={ task.completedButtonText }
+							completedTitle={ task.completedTitle }
+							description={ task.description }
+							duration={ task.duration }
+							key={ task.id }
+							onClick={ this.handleTaskStart( task ) }
+							onDismiss={ this.handleTaskDismiss( task ) }
+							title={ task.title }
+						/>
+					) ) }
+				</Checklist>
 			</Fragment>
 		);
 	}
 }
 
 const mapStateToProps = state => {
 	const siteId = getSelectedSiteId( state );
-	const siteSlug = getSiteSlug( state, siteId );
-	const siteChecklist = getSiteChecklist( state, siteId );
-	const isJetpack = isJetpackSite( state, siteId );
-	const tasks = isJetpack ? jetpackTasks : wpcomTasks;
-	const tasksFromServer = get( siteChecklist, [ 'tasks' ] );
 
 	return {
+		isJetpack: isJetpackSite( state, siteId ),
 		siteId,
-		siteSlug,
-		tasks: tasksFromServer ? mergeObjectIntoArrayById( tasks, tasksFromServer ) : null,
```

Comment:
Sorry for this but we can't just delete the line of code since the response from the server-side can return extra information besides whether a task completes or not. Let me give you an example. The URL of the very first post can be changed based on themes. So the server-side finds and returns its URL then this `mergeObjectIntoArrayById` applies the new URL to the corresponding task. Here's a link to the server-side code: 1efb3-pb

You don't need to do anything. I will handle this problem in my PR: https://github.com/Automattic/wp-calypso/pull/26721

Reply:
Ugh, I missed that in my review. Sry.